packages/jaeger-ui/src/model/trace-viewer.js

```diff
@@ -14,6 +14,6 @@
 
 // eslint-disable-next-line import/prefer-default-export
 export function getTraceName(spans) {
-  const span = spans.filter(sp => !sp.references || !sp.references.length)[0];
+  const span = spans.find(sp => !sp.references || !sp.references.some(ref => ref.traceID === sp.traceID));
```

Comment:
This doesn't seem like an equivalent or a superset of the previous logic.

I would rather fix this like this:
  * find all spans that either have no parent reference, or a parent reference that is not found in the trace
  * if only one such span is found, it's the root (i.e. source of "trace name")
  * if more than one such span is found, then
    * if exactly one of them has no parent refs, it's the root
    * otherwise give up (or, could still do something like pick the one with the earliest start time)

Clearly, we need a set of unit tests for this logic.